internal/authz/oidc.go

```diff
@@ -396,6 +396,81 @@ func (o *oidcHandler) retrieveTokens(ctx context.Context, log telemetry.Logger,
 	setDenyResponse(resp, deny, codes.Unauthenticated)
 }
 
+// buildAuthHeader builds the authorization header for the client according to the
+// client authentication method specified in the OIDCConfig.
+//
+// The function returns an error if the client authentication method is unspecified
+// or if the implementation for the specified method is not supported.
+func buildAuthHeader(config *oidcv1.OIDCConfig) (http.Header, error) {
+
+	headers := http.Header{}
```

Comment:
There is no need to allocate something here if it is going to be replaced.
```suggestion
	var headers http.Header
```